test/cost_estimator.hpp

```diff
@@ -11,13 +11,17 @@ namespace arbitration_graphs_tests {
 using namespace arbitration_graphs;
 
 
-struct CostEstimatorFromCostMap : public CostEstimator<DummyCommand> {
+struct CostEstimatorFromCostMap : public CostEstimator<DummyEnvironmentModel, DummyCommand> {
     using CostMap = std::map<DummyCommand, double>;
 
     CostEstimatorFromCostMap(const CostMap& costMap, const double activationCosts = 0)
             : costMap_{costMap}, activationCosts_{activationCosts} {};
 
-    virtual double estimateCost(const DummyCommand& command, const bool isActive) override {
+    virtual double estimateCost(const Time& /*time*/,
+                                const DummyEnvironmentModel& environmentModel,
+                                const DummyCommand& command,
+                                const bool isActive) override {
+        environmentModel.getObservation();
```

Comment:
A comment explaining this "simulation" would improve the understandability for users a lot. Same goes for the `getObservation()` calls in `test/dummy_types.hpp`

I guess sth. like 
> normally you would use the env model here to estimate the cost / determine applicability (invocation) / …